addons/sale/migrations/10.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -1,10 +1,19 @@
 ---Fields in module 'sale'---
 sale         / account.invoice          / partner_shipping_id (many2one): NEW relation: res.partner
 sale         / account.invoice.line     / layout_category_id (many2one) : NEW relation: sale.layout_category
+sale         / product.pricelist        / discount_policy (selection)   : NEW selection_keys: ['with_discount', 'without_discount']
+sale         / product.template         / expense_policy (selection)    : NEW selection_keys: ['cost', 'no', 'sales_price']
 # NOTHING TO DO
+
 sale         / product.template         / invoice_policy (selection)    : selection_keys is now '['delivery', 'order']' ('['cost', 'delivery', 'order']')
-# TODO: pre-migration: copy column
-# TODO: post-migration: map value 'cost' to new value
+# DONE: pre-migration: copy column
+# DONE: post-migration: map value 'cost' to new value
+
+sale         / product.template         / sale_line_warn (selection)    : NEW required: required, selection_keys: ['block', 'no-message', 'warning'], req_default: no-message
+sale         / product.template         / sale_line_warn_msg (text)     : NEW 
+sale         / res.partner              / sale_warn (selection)         : NEW required: required, selection_keys: ['block', 'no-message', 'warning'], req_default: no-message
+sale         / res.partner              / sale_warn_msg (text)          : NEW 
+# NOTHING TO DO
```

Comment:
Please check it if warning module was installed on v9 and rescue values from it.